desc/coils.py

```diff
@@ -187,10 +270,16 @@ def _compute_position(self, params=None, grid=None, **kwargs):
             x = x.at[:, :, 1].set(jnp.mod(x[:, :, 1], 2 * jnp.pi))
         return x
 
-    def compute_magnetic_field(
-        self, coords, params=None, basis="rpz", source_grid=None, transforms=None
+    def _compute_A_or_B(
```

Comment:
See comments on the subclasses, but if this is getting superseded by child classes then it should be an abstract method. Same for `compute_magnetic_field` and `compute_magnetic_vector_potential`

Reply:
the point is to avoid duplicated code. A lot of the preprocessing is the same, the only thing that really changes is the integrand

Reply:
There are only about a dozen lines of duplicate code, and that is mostly just for checking "rpz" vs "xyz" basis which we already do in all the compute functions too. I would rather make a single util function to do that check and conversion that we can call everywhere. Because the header/docstring of this function alone is like 40 lines of code that also gets duplicated many times, even if it isn't "real code".

Reply:
for `SplineMagneticField` and `CurrentPotentialField` the duplicated code is 100+ lines

I don't have a strong preference either way but I do think this current way is cleaner for avoiding duplicated code, and this is all under the hood stuff anyways. Unless there is a stronger case for splitting this up I want to keep it as it is now.

Reply:
In other words, move the actual duplicate code to a separate util function, and then all the code that is specific to A vs B can be moved to their respective compute functions.